### ISSUE TITLE
Cavity construction can be extremely slow

### ISSUE THREAD

**Issue by shoefener** [PCMSolver/pcmsolver#26]
The input file

```
UNITS = AU
CAVITY
{
                TYPE = GEPOL
                MODE = EXPLICIT
             SPHERES = [0.0,0.0,0.0,12.0] 
}
MEDIUM
{
                SOLVENT = WATER
}
```

can lead to extremely long computation times (about 20 minutes and more) for the cavity construction 

```
pcm_context = pcmsolver_new( PCMSOLVER_READER_OWN,nr_nuclei,charges, &
                                  & coordinates,symmetry_info,host_input         )

```

on some computers, while on others it takes about 1 second only.

**Comment by robertodr** [comment]
Can you be more specific about the meaning of "on some computers"? Is the library compiled in release mode in all cases? Do you have a time profile? It would be extremely helpful if you could provide the following:
1. The time elapsed in the call to `pcmsolver_new` and the time elapsed in the call to `pcmsolver_get_centers`
2. If you compile the library with `-DENABLE_TIMER=ON` and call `pcmsolver_write_timings` before cleaning up the PCM context object, various internal timings should be printed to a file called `pcmsolver.timer.dat`. Could you post this file?
I am in the process of preparing the 1.1.1 patch of the library and I will insert some more timing checkpoints. I suggest you wait a couple of days before doing the things I suggested, so that we are working on the same exact version.

**Comment by shoefener** [comment]
Unfortunately, I cannot be more specific on "some computers" since I could not derive a general rule when this occurs.. I compiled the library once and use the same binary for all my KOALA compilations. For the sake of curiosity, what would be the grid construction time you expect for this input? 

However, I will do what you suggested once you have the 1.1.1 version in place.

**Comment by robertodr** [comment]
@shoefener I apologize for getting back to you on this so late. I am still struggling to get the new version out... From the few experiments I've run, I would say that the bottleneck is not cavity formation, but system matrix construction and/or inversion. Could you try with a larger `area` input parameter, say 1.0 angstrom^2 instead of the default value of 0.3 au^2? For the type of systems I think you're interested in, a coarser cavity discretization should be fine.

**Comment by robertodr** [comment]
The problem was actually in the inversion of the PCM matrix. Solved with 34d2fea
